pkg/vulnsrc/oracle-oval/oracle-oval.go

```diff
@@ -156,21 +168,118 @@ func (vs *VulnSrc) commit(tx *bolt.Tx, ovals []OracleOVAL) error {
 				Severity:    severityFromThreat(oval.Severity),
 			}
 
-			err := vs.Put(tx, PutInput{
+			input := PutInput{
 				VulnID:     vulnID,
 				Vuln:       vuln,
-				Advisories: advisories,
-				OVAL:       oval,
-			})
-			if err != nil {
-				return xerrors.Errorf("db put error: %w", err)
+				Advisories: maps.Clone(advisories),
+				OVALs:      []OracleOVAL{oval},
+			}
+
+			if savedInput, ok := putInputs[input.VulnID]; ok {
+				input.OVALs = append(input.OVALs, savedInput.OVALs...)
+
+				for inputPkg, inputAdvs := range input.Advisories {
+					if savedPkgAdvs, pkgFound := savedInput.Advisories[inputPkg]; pkgFound {
+						inputAdvs.Entries = append(savedPkgAdvs.Entries, inputAdvs.Entries...)
+					}
+					savedInput.Advisories[inputPkg] = inputAdvs
+				}
+				input.Advisories = savedInput.Advisories
 			}
+			putInputs[input.VulnID] = input
+		}
+	}
+
+	for _, input := range putInputs {
+		for pkg, advs := range input.Advisories {
+			input.Advisories[pkg] = resolveAdvisoriesEntries(advs)
+		}
+
+		err := vs.Put(tx, input)
+		if err != nil {
+			return xerrors.Errorf("db put error: %w", err)
 		}
 	}
 
 	return nil
 }
 
+// resolveAdvisoriesEntries removes entries with the same fixedVersion.
+// Additionally, it only selects the latest fixedVersion for each flavor.
+func resolveAdvisoriesEntries(advisories types.Advisories) types.Advisories {
+	fixedVersions := lo.Map(advisories.Entries, func(entry types.Advisory, _ int) string {
+		return entry.FixedVersion
+	})
+	fixedVer, resolvedVers := resolveVersions(fixedVersions)
+	entries := lo.Map(resolvedVers, func(ver string, _ int) types.Advisory {
+		return types.Advisory{
+			FixedVersion: ver,
+		}
+	})
+	return types.Advisories{
+		FixedVersion: fixedVer,
+		Entries:      entries,
+	}
+}
+
+// resolveVersions removes duplicates and returns normal flavor + only one version for each flavor.
+func resolveVersions(vers []string) (string, []string) {
+	vers = lo.Uniq(vers)
+
+	fixedVers := make(map[PkgFlavor]string)
+	for _, ver := range vers {
+		flavor := PackageFlavor(ver)
+		if savedVer, ok := fixedVers[flavor]; ok {
+			v := version.NewVersion(ver)
+			sv := version.NewVersion(savedVer)
+			if v.LessThan(sv) {
+				ver = savedVer
+			}
+		}
+		fixedVers[flavor] = ver
+	}
+
+	versions := lo.Values(fixedVers)
+	slices.Sort(versions)
+
+	fixedVersion := fixedVers[NormalPackageFlavor]
+
+	return fixedVersion, versions
+}
+
+type PkgFlavor string
+
+const (
+	NormalPackageFlavor   PkgFlavor = "normal"
+	FipsPackageFlavor     PkgFlavor = "fips"
+	Ksplice1PackageFlavor PkgFlavor = "ksplice1"
+	Ksplice2PackageFlavor PkgFlavor = "ksplice2"
```

Comment:
We don't use `ksplice1` flavor for `ksplice2` (and vice versa) in Trivy:
https://github.com/aquasecurity/trivy/blob/dd0a64a1cf0cd76e6f81e3ff55fa6ccb95ce3c3d/pkg/detector/ospkg/oracle/oracle.go#L72-L77

So i separated `ksplice1` and `ksplice2`.
After these changes db size increased to `0.9MB`:
```
➜  du -sh ./assets-with-flavors
539M	./assets-with-flavors
➜  du -sh ./assets             
540M	./assets 
```

Reply:
It's been long enough that I've lost track of all the details on this, but I did want to pass along some information I got from @tvierling (from the Oracle Linux support/development group) when I was working though this.  I won't quote his email directly without his permission, but the gist of it is that `ksplice` is the flavor, and the number after `ksplice` is just a revision / build number (which should be considered.. `x.y.z.ksplice2` > `x.y.z.ksplice1`).  So I don't think you want to separate out as different flavors.

Reply:
I'm glad you're still here and helping with this task.
Sorry we're working on this after so long...

Also, I want to thank you for your work! Our work with you helped me remember the details and update this PR.

> but the gist of it is that ksplice is the flavor, and the number after ksplice is just a revision / build number (which should be considered.. x.y.z.ksplice2 > x.y.z.ksplice1). So I don't think you want to separate out as different flavors.

I originally used your `PackageFlavor` function. 
But I was confused that we separate ksplice1 and ksplice2 in Trivy and i updated PR.

But since the person from Oracle Linux says that we shouldn't divide them, then there is a mistake in our logic Trivy.

Thanks! I will revert last commit.

Reply:
done - https://github.com/aquasecurity/trivy-db/pull/221/commits/a135499d61b191053c033a5dc22ef2dd24e244b5